### DISCUSSION TITLE
Dataframe으로 만들 때 특정 Table Code를 하드코딩한 이유가 뭔가요?

### DISCUSSION THREAD

**Discussion by bhnvx** [josw123/dart-fss#173] [General]
`table = self.parent.get_table_by_code('d999004')`
`unit = get_value_from_dataset(table.cls, table.dataset, 'dart-gcd_EntityReportingCurrencyISOCode', ignore_case=True)`

위 처럼 코드가 작성된걸 확인 했는데, 이유가 뭔지 궁금해서 글 남깁니다.

추가적으로 to_DataFrame() 로 바꾸는 데이터는 재무제표만 바뀌는 부분일까요?
다트 전자공시에서 xbrl viewer로 보이는 데이터랑 차이가 있어보여서요 !

**Comment by josw123** [comment]
안녕하세요.

1.
하드코딩된 부분은 XBRL 데이터 중 "기업정보-공시대상정보-통화ISO코드" 부분으로 일부 기업의 경우 KWR 이 아닌 CNY 등 다른 나라의 통화를 사용하는 경우가 있습니다. 이를 참고하여 처리하기 위해 통화ISO코드 부분을 참고하여 처리하게 됩니다.

2.
얼마전까지만 하더라도 주석부분이 XBRL로 제공되지 않았기 때문에 재무제표 부분만 처리되도록 되어있습니다.
다트 전자공시 부분의 XBRL viewer 부분과 같이 처리 하기 위해서는 업데이트가 필요한 상황입니다.
문제는 알고 있으나 현재 현업이 좀 많이 바빠서 처리 못하고 있는 중입니다.

**Reply by bhnvx** [reply]
답변 감사드립니다!

**Reply by CHOKYUNGCHO** [reply]
안녕하세요? 다트 전자공시 부분의 XBRL viewer 부분과 같이 처리 하기 위해서는 업데이트가 필요한 상황입니다.  어떤 부분을 업데이트를 해야할까요?

**Reply by josw123** [reply]
기존 XBRL 데이터에는 대부분 단순 문자열 혹은 숫자만 입력되어 있었습니다. 그러나 주석이 추가되면서 추가적인 문장이나 문서 형태 등이 추가된 것으로 생각되며 이에 관련된 업데이트가 필요한 상황입니다.
